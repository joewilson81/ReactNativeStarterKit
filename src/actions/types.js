export const BUTTON_PUSH = 'button_push';
