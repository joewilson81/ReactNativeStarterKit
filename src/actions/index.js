import { BUTTON_PUSH } from './types';

export const buttonPush = () => {
  return { type: BUTTON_PUSH };
}
