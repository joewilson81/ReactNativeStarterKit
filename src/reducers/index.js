import { combineReducers } from 'redux';
import { BUTTON_PUSH } from '../actions/types';

const INITIAL_STATE = { buttonPressed: false };

export default combineReducers({
  dummy: (state = INITIAL_STATE, action) => {
    switch (action.type) {
      case BUTTON_PUSH:
        return { ...state, buttonPressed: true };

      default:
        return state;
    }
  }
});
