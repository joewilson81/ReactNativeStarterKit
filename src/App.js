import React, { Component } from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import { View, Text, StyleSheet } from 'react-native';
import reducers from './reducers';

class App extends Component {
  render() {
    const middleware = [];

    // Only apply the following middleware in development!
    if (process.env.NODE_ENV !== 'production') {
      middleware.push(require('redux-logger')());
    }

    const store = createStore(reducers, {}, applyMiddleware(...middleware));
    const { containerStyle, welcomeStyle } = styles;

    return (
      <Provider store={store}>
        <View style={containerStyle}>
          <Text style={welcomeStyle}>
            Welcome to React Native!
          </Text>
        </View>
      </Provider>
    );
  }
}

const styles = StyleSheet.create({
  containerStyle: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcomeStyle: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  }
});


export default App;
