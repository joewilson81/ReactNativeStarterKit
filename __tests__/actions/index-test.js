import * as actions from '../../src/actions';

describe('actions', () => {
  it('should create an action to push a button', () => {
    expect(actions.buttonPush()).toMatchSnapshot();
  });
});
