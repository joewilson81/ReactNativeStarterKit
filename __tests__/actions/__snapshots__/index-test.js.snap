// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should create an action to push a button 1`] = `
Object {
  "type": "button_push",
}
`;
